clock_exercise/clockmanager.h

```diff
@@ -0,0 +1,41 @@
+#ifndef CLOCKMANAGER_H
+#define CLOCKMANAGER_H
+
+#include <QObject>
+#include <QTimer>
+#include <memory>
+
+class ClockManager : public QObject
+{
+    Q_OBJECT
+public:
+    explicit ClockManager(QObject *parent = nullptr);
+
+signals:
+    void dateTextChanged();
+    void minutesClockHandAngleChanged();
+    void hoursClockHandsAngleChanged();
+
+public slots:
+
+private:
+    void refreshClockHandsAngles();
+    void refreshDateText();
+
+    std::unique_ptr<QTimer> mTimerForRefresh;
```

Comment:
Qui `std::unique_ptr` non ti serve.
QTimer è un QObject, e come tutti i QObject può ricevere un puntatore al `parent` nel costruttore nel caso tu voglia che venga chiamata la `delete` sull'oggetto alla distruzione del `parent`.

Detto questo, secondo me non serve nemmeno avere avere puntatori in questo caso: se tanto sappiamo che QTimer dovrà vivere per tutto il tempo in cui lo farà ClockManager, tanto vale averlo allocato nella memoria locale di ClockManager. Ricordati che in questo caso il `parent` che gli passi nel costruttore deve essere un `nullptr`.